File: src/main/java/upbrella/be/rent/repository/RentRepository.java
package upbrella.be.rent.repository;

import org.springframework.data.jpa.repository.JpaRepository;

import java.util.Optional;
import upbrella.be.rent.entity.History;

public interface RentRepository extends JpaRepository<History, Long>, RentRepositoryCustom {

    Optional<History> findByUserIdAndReturnedAtIsNull(Long userId);

    long countByRentStoreMetaId(long storeId);

    long countAllByReturnedAtIsNotNullAndPaidAtIsNotNullAndRefundedAtIsNull();
}

File: src/main/kotlin/upbrella/be/rent/repository/RentRepository.kt
package upbrella.be.rent.repository

import org.springframework.data.jpa.repository.JpaRepository
import upbrella.be.rent.entity.History
import java.util.Optional

interface RentRepository : JpaRepository<History, Long>, RentRepositoryCustom {
    fun findByUserIdAndReturnedAtIsNull(userId: Long): Optional<History>

    fun countByRentStoreMetaId(storeId: Long): Long

    fun countAllByReturnedAtIsNotNullAndPaidAtIsNotNullAndRefundedAtIsNull(): Long
}